### PR TITLE
fix(client): put /update-email behind authentication

### PR DESCRIPTION
**Pull request by huyenltnguyen**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

This PR:
- Updates the Update Email page to block access if the user is not logged in
- Is an alternative to #52571. The PR adds the signin check directly to `client/src/pages/update-email.tsx`, while mine converts the page to a client-only-route
  - Related discussion: https://github.com/freeCodeCamp/freeCodeCamp/pull/52571#discussion_r1500239514
- Closes #52568

**Note for reviewers:** The PR involves moving files around as well as renaming them. Reviewing by commits is recommended.

## Result

https://github.com/freeCodeCamp/freeCodeCamp/assets/25715018/369883c0-a2ea-421b-853b-abaa7ded4431

### CODE REVIEW

**Review by huyenltnguyen — COMMENTED** [review]
Tests passed locally but failed in CI. I'll dig deeper 🥲 

Screenshot 2024-03-18 at 15 08 25

**Review by raisedadead — APPROVED** [review]

File: e2e/update-email.spec.ts [hunk 1]
```diff
@@ -38,26 +29,43 @@ test.describe('The update-email page', () => {
...
+  test.beforeEach(async ({ page }) => {
+    await page.goto('/update-email');
+  });
+
+  test('should redirect to the signin page', async ({ page }) => {
+    if (process.env.CI) {
+      await page.waitForURL('**/learn/');
+    } else {
+      await page.waitForURL('**/learn');
+    }
```
Comment (ojeytonwilliams) [inline e2e/update-email.spec.ts:65]:
```suggestion
    await page.waitForURL(/\/learn\/?/);
```
I'm in two minds about this if this is an improvement, but it is a little more concise :shrug:

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Nice work, @huyenltnguyen!

I think we can tidy up some stuff while we're here:

File: e2e/update-email.spec.ts [hunk 2]
```diff
@@ -38,26 +29,47 @@ test.describe('The update-email page', () => {
...
+  });
+
+  test('should sign the user in and redirect them to /learn', async ({
+    page,
+    browserName
+  }) => {
+    // The signin step involves multiple navigations, which results a network error in Firefox.
+    // The error is harmless but Playwright doesn't suppress it, causing the test to fail.
+    // Ref: https://github.com/microsoft/playwright/issues/20749
+    test.skip(browserName === 'firefox');
```
[the remark this answers is not in the document]
Comment (ojeytonwilliams) [reply e2e/update-email.spec.ts:67]: It seems fine to skip for now. Thanks for looking into this, @huyenltnguyen

**Review by ojeytonwilliams — APPROVED** [review]
Nice work, @huyenltnguyen!

LGTM :+1:

**Pull request merged**